homeassistant/components/abode/config_flow.py

```diff
@@ -102,15 +102,7 @@ async def _async_create_entry(self) -> ConfigFlowResult:
         existing_entry = await self.async_set_unique_id(self._username)
 
         if existing_entry:
-            self.hass.config_entries.async_update_entry(
-                existing_entry, data=config_data
-            )
-            # Reload the Abode config entry otherwise devices will remain unavailable
-            self.hass.async_create_task(
-                self.hass.config_entries.async_reload(existing_entry.entry_id)
-            )
-
-            return self.async_abort(reason="reauth_successful")
+            return self.async_update_reload_and_abort(existing_entry, data=config_data)
```

Comment:
It seems that the reauth process can create a new config entry.
Is this intentional?